### ISSUE TITLE
ValueError: invalid version number '2.7.0rc3'

### ISSUE THREAD

**Issue by ghisvail** [h5py/h5py#834]
See logs in [travis](https://travis-ci.org/h5py/h5py/jobs/195097900) or [Debian](http://debomatic-amd64.debian.net/distribution#unstable/h5py/2.7.0~rc3-1/buildlog) builds.

**Comment by aragilar** [comment]
distutils' version parsing support is not PEP 440 compliant (see https://www.python.org/dev/peps/pep-0386/#id15), only a and b are treated as prereleases (e.g. 1.0a5 and 5.2b6,). This will work fine in the actual release. It'd be better to make the tuple the canonical source rather than the string, which avoids this in the future.

**Comment by ghisvail** [comment]
> only a and b are treated as prereleases (e.g. 1.0a5 and 5.2b6,)

I thought `c` was supposed to be supported for release candidates?

**Comment by ghisvail** [comment]
Answering my own question, it does not.
```python
>>> from distutils.version import StrictVersion as V
>>> ver = V('2.7.0rc3')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.5/distutils/version.py", line 40, in __init__
    self.parse(vstring)
  File "/usr/lib/python3.5/distutils/version.py", line 137, in parse
    raise ValueError("invalid version number '%s'" % vstring)
ValueError: invalid version number '2.7.0rc3'
>>> ver = V('2.7.0c3')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.5/distutils/version.py", line 40, in __init__
    self.parse(vstring)
  File "/usr/lib/python3.5/distutils/version.py", line 137, in parse
    raise ValueError("invalid version number '%s'" % vstring)
ValueError: invalid version number '2.7.0c3'
```

**Comment by ghisvail** [comment]
I am currently testing `rc3` + cherry-picked fixes for #833 and #835
